module/VuFindConsole/src/VuFindConsole/Command/OnlinePayment/MonitorCommand.php

```diff
@@ -0,0 +1,378 @@
+<?php
+
+/**
+ * Console service for processing unregistered online payments.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) The National Library of Finland 2016-2025.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Service
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     http://vufind.org/wiki/vufind2:developer_manual Wiki
+ */
+
+namespace VuFindConsole\Command\OnlinePayment;
+
+use Laminas\View\Renderer\PhpRenderer;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\ConsoleOutputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use VuFind\Config\Feature\EmailSettingsTrait;
+use VuFind\Db\Entity\PaymentEntityInterface;
+use VuFind\Db\Service\AuditEventServiceInterface;
+use VuFind\Db\Service\PaymentServiceInterface;
+use VuFind\Db\Type\AuditEventSubtype;
+use VuFind\Mailer\Mailer;
+use VuFind\OnlinePayment\OnlinePaymentEventTrait;
+use VuFind\OnlinePayment\OnlinePaymentManager;
+
+use function count;
+
+/**
+ * Console service for processing unregistered online payments.
+ *
+ * @category VuFind
+ * @package  Service
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     http://vufind.org/wiki/vufind2:developer_manual Wiki
+ */
+#[AsCommand(
+    name: 'online_payment/monitor'
```

Comment:
For completeness, we should add this new command somewhere in the console menu, as per #4613. It probably fits in the "Data Management Tools" submenu along with things like scheduledsearch/notify.